leveldown.js

```diff
@@ -61,15 +60,20 @@ function SubDown (db, prefix, opts) {
   this.prefix = this.ownPrefix
   this._beforeOpen = opts.open
 
+  // TODO: assert that prefix sorts after "ceiling"
+  var ceiling = String.fromCharCode(separator.charCodeAt(0) + 1)
   var self = this
 
   this._wrap = {
     gt: function (x) {
       return concat(self.prefix, x || '', true)
     },
     lt: function (x) {
-      if (Buffer.isBuffer(x) && !x.length) x = END
-      return concat(self.prefix, x || '\xff')
+      if (!x || isEmptyBuffer(x)) {
+        return self.prefix.slice(0, -1) + ceiling
```

Comment:
An example: if the separator is `!` then the "ceiling" is `"`. If you have a bunch of sublevels with prefixes `['!a!', '!a!!a!', '!aa!', '!b!']` and you want to get everything in `!a!`, then that's everything less than `!a"`.

```
$ node
> ['!a!', '!a!!a!', '!aa!', '!b!', '!a"'].sort()
[ '!a!', '!a!!a!', '!a"', '!aa!', '!b!' ]
```

Is there a case where this doesn't work? Apart from a custom `separator` that leaves no room for the ceiling, like `sub(db, '0', { separator: '/' })`.

Reply:
this is a really good idea also for defining keys in general cases and get a range of it.